assignment/assignment.hs

```diff
@@ -26,25 +31,39 @@ import Data.Char(intToDigit)
 -- predicate returns `True`. Implement `myfilter` using `foldl`. (Hint: you 
 -- might need to use `++` rather than `:` to build the result)
 
--- myfilter :: (a -> Bool) -> [a] -> [a]
+--myfilter :: (a -> Bool) -> [a] -> [a]
+
+addtolist fn list element
+  | (fn element == True)  = list : element
+  | otherwise             = list
 
```

Comment:
Yup, except....

This is a bugbear I have with code in any language:

Why write `(fn element == True)`? If `fn` returns `True`, then it is already true :)

Now in Haskell this is just redundant. In other languages, though, it's a bug. In JavaScript, for example, the number 99 is true in a conditional context, so

``` js
if (fnReturning99()) console.log("true") else console.log("false")
if (fnReturning99() === true) console.log("true") else console.log("false")
```

will output `true, false`...